commit/chainfee/outcome.go

```diff
@@ -0,0 +1,185 @@
+package chainfee
+
+import (
+	"fmt"
+	"sort"
+	"time"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/types"
+	cciptypes "github.com/smartcontractkit/chainlink-common/pkg/types/ccipocr3"
+
+	"github.com/smartcontractkit/chainlink-ccip/internal/libs/mathslib"
+
+	"github.com/smartcontractkit/chainlink-ccip/internal/plugincommon"
+	"github.com/smartcontractkit/chainlink-ccip/internal/plugincommon/consensus"
+)
+
+func (p *processor) getConsensusObservation(
+	aos []plugincommon.AttributedObservation[Observation],
+) (Observation, error) {
+	aggObs := aggregateObservations(aos)
+
+	// consensus on the fChain map uses the role DON F value
+	// because all nodes can observe the home chain.
+	donThresh := consensus.MakeConstantThreshold[cciptypes.ChainSelector](consensus.TwoFPlus1(p.fRoleDON))
+	fChains := consensus.GetConsensusMap(p.lggr, "fChain", aggObs.FChain, donThresh)
+
+	fDestChain, exists := fChains[p.destChain]
+	if !exists {
+		return Observation{},
+			fmt.Errorf("no consensus value for fDestChain, destChain: %d", p.destChain)
+	}
+	if len(aggObs.Timestamps) < fDestChain {
```

Comment:
I don't think this is the right way to check for consensus one the number of observations - is `Timestamps` simply the `time.Now().UTC()` stuff that each node observes? I think it should be safe to assert that `len(aggObs.Timestamps) > consensus.TwoFPlus1(fDestChain)`

Reply:
Yes, good call 👍

Reply:
Ah shoot I meant `len(aggObs.Timestamps) >= consensus.TwoFPlus1(fDestChain)` (notice >= not >)

Reply:
I'm have the condition reversed and using `<` so should be okay